### PR TITLE
Implement fiam and fiamd Kotlin library.

### PR DESCRIPTION
**Pull request by rlazo**

### CODE REVIEW

File: firebase-inappmessaging-display/ktx/src/test/kotlin/com/google/firebase/inappmessaging/display/ktx/InAppMessagingDisplayTests.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+@RunWith(RobolectricTestRunner::class)
+class InAppMessagingDisplayTests : BaseTestCase() {
+
+    @Test
+    fun `inAppMessagingDisplay should delegate to FirebaseInAppMessagingDisplay#getInstance()`() {
+        assertThat(Firebase.inAppMessagingDisplay).isSameInstanceAs(FirebaseInAppMessagingDisplay.getInstance())
+    }
+//
+//    @Test
+//    fun `FirebaseApp#functions should delegate to FirebaseFunctions#getInstance(FirebaseApp)`() {
```
Comment (vkryachko) [inline firebase-inappmessaging-display/ktx/src/test/kotlin/com/google/firebase/inappmessaging/display/ktx/InAppMessagingDisplayTests.kt]: Is this intended? If not pls remove

Comment (rlazo) [reply firebase-inappmessaging-display/ktx/src/test/kotlin/com/google/firebase/inappmessaging/display/ktx/InAppMessagingDisplayTests.kt]: That's... not how it's suppose to look :-) fixed.

File: firebase-inappmessaging-display/ktx/src/main/kotlin/com/google/firebase/inappmessaging/display/InAppMessagingDisplay.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+// /** Returns the [FirebaseFunctions] instance of a given [region]. */
+// fun Firebase.functions(region: String): FirebaseFunctions = FirebaseFunctions.getInstance(region)
+
+// /** Returns the [FirebaseFunctions] instance of a given [FirebaseApp]. */
+// fun Firebase.functions(app: FirebaseApp): FirebaseFunctions = FirebaseFunctions.getInstance(app)
+
+// /** Returns the [FirebaseFunctions] instance of a given [FirebaseApp] and [region]. */
+// fun Firebase.functions(app: FirebaseApp, region: String): FirebaseFunctions =
+//         FirebaseFunctions.getInstance(app, region)
```
Comment (rosariopfernandes) [inline firebase-inappmessaging-display/ktx/src/main/kotlin/com/google/firebase/inappmessaging/display/InAppMessagingDisplay.kt]: Don't forget to remove these comments before merging :sweat_smile:

Comment (rlazo) [reply firebase-inappmessaging-display/ktx/src/main/kotlin/com/google/firebase/inappmessaging/display/InAppMessagingDisplay.kt]: [me] Looks around and sweeps it under the carpet...

File: firebase-inappmessaging-display/ktx/src/main/kotlin/com/google/firebase/inappmessaging/display/InAppMessagingDisplay.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import androidx.annotation.Keep
+import com.google.firebase.FirebaseApp
+import com.google.firebase.inappmessaging.display.FirebaseInAppMessagingDisplay
+import com.google.firebase.components.Component
+import com.google.firebase.components.ComponentRegistrar
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.platforminfo.LibraryVersionComponent
+
+/** Returns the [FirebaseFunctions] instance of the default [FirebaseApp]. */
```
Comment (vkryachko) [inline firebase-inappmessaging-display/ktx/src/main/kotlin/com/google/firebase/inappmessaging/display/InAppMessagingDisplay.kt]:
```suggestion
/** Returns the [FirebaseInAppMessagingDisplay] instance of the default [FirebaseApp]. */
```

File: firebase-inappmessaging/ktx/src/main/kotlin/com/google/firebase/inappmessaging/ktx/InAppMessaging.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import androidx.annotation.Keep
+import com.google.firebase.FirebaseApp
+import com.google.firebase.inappmessaging.FirebaseInAppMessaging
+import com.google.firebase.components.Component
+import com.google.firebase.components.ComponentRegistrar
+
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.platforminfo.LibraryVersionComponent
+
+/** Returns the [FirebaseFunctions] instance of the default [FirebaseApp]. */
```
Comment (vkryachko) [inline firebase-inappmessaging/ktx/src/main/kotlin/com/google/firebase/inappmessaging/ktx/InAppMessaging.kt]:
```suggestion
/** Returns the [FirebaseInAppMessaging] instance of the default [FirebaseApp]. */
```

File: firebase-inappmessaging/ktx/src/test/kotlin/com/google/firebase/inappmessagging/ktx/InAppMessaggingTests.kt [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @After
+    fun cleanUp() {
+        FirebaseApp.clearInstancesForTest()
+    }
+}
+
+@RunWith(RobolectricTestRunner::class)
+class FunctionsTests : BaseTestCase() {
```
Comment (vkryachko) [inline firebase-inappmessaging/ktx/src/test/kotlin/com/google/firebase/inappmessagging/ktx/InAppMessaggingTests.kt]:
```suggestion
class FirebaseInAppMessagingTests : BaseTestCase() {
```

**Pull request merged**